File: producer-consumer/src/main/scala/com/wardziniak/kafka/scala/config/ConfigBuilder.scala
package com.wardziniak.kafka.scala.config

import java.util.{Optional, Properties}

import com.wardziniak.kafka.config.ConfigBuilder

trait ConfigBuilder {

  private[config] val BOOTSTRAP_SERVERS_DEFAULT = "localhost:9092"
  protected var bootstrapServers: Option[String] = Option.empty

  def withServer(bootstrapServer: String): ConfigBuilder = {
    this.bootstrapServers = Option(bootstrapServer)
    this
  }

  def buildConfig: Properties
}

File: kafka-utils/src/main/java/com/wardziniak/kafka/config/ConfigBuilder.java
package com.wardziniak.kafka.config;

import java.util.Optional;
import java.util.Properties;

/**
 * Created by wardziniak on 09.06.18.
 */
public abstract class ConfigBuilder {

    static final String BOOTSTRAP_SERVERS_DEFAULT = "localhost:9092";
    protected Optional<String> bootstrapServers = Optional.empty();

    public ConfigBuilder withServer(String bootstrapServer) {
        this.bootstrapServers = Optional.ofNullable(bootstrapServer);
        return this;
    }

    public abstract Properties buildConfig();
}
